docker/docker_client.go

```diff
@@ -187,45 +187,55 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 	}, nil
 }
 
+type authScope struct {
+	remoteName string
+	actions    string
+}
```

Comment:
Actually, looking at this more, the `authScope` is fixed for the life of a `dockerClient`, and `remoteName` can be computed from a `dockerReference`; so can we set the scope in `newDockerClient` and drop it from `makeRequest`* entirely?  Sorry for not noticing this earlier.

Would then having the `requestOptions` still be worthwhile?